src/ethereum_test_tools/common/types.py

```diff
@@ -1688,7 +1695,7 @@ def with_signature_and_sender(self) -> "Transaction":
                 tx.v += 27
 
         # Remove the secret key because otherwise we might attempt to sign again (?)
-        tx.secret_key = None
+        # tx.secret_key = None
```

Comment:
We should simply remove these two lines instead of commenting them.

Reply:
This also makes a lot of tests fail (see `tox -e framework`), so we might need to remove this check: https://github.com/ethereum/execution-spec-tests/blob/641470d21e94adf2fe6656800b52d8dc194c4a06/src/ethereum_test_tools/common/types.py#L1334

Reply:
If I copy it to another field then is no problem

Reply:
changed to use a separate field so not to mess any existing logic (as it fails if I comment this line)